src/main/java/bro/Parser.java

```diff
@@ -0,0 +1,71 @@
+package bro;
+
+import bro.command.*;
+import bro.task.Deadline;
+import bro.task.Event;
+import bro.task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Parser {
+
+    public Command parse(String str) throws BroException {
+
+        this.checkEmptyInput(str);
+        String[] in = str.split(" ");
+        String input = in[0];
+            switch (input) {
+                case "list":
+                    return new ListCommand();
+                case "mark":
+                    return new ModifyCommand(ModifyCommand.ModifyType.MARK, Integer.parseInt(in[1].trim()));
+                case "unmark":
+                    return new ModifyCommand(ModifyCommand.ModifyType.UNMARK, Integer.parseInt(in[1].trim()));
+                case "todo":
+                    return new AddCommand(new Todo(in[1]));
+                case "deadline":
+                    return new AddCommand(new Deadline(in[1], str.split("/by")[1].trim()));
+                case "event":
+                    return new AddCommand(new Event(in[1], str.split("/at")[1].trim()));
+                case "delete":
+                    return new DeleteCommand(Integer.parseInt(in[1].trim()));
+                case "bye":
+                    return new ExitCommand();
+                default:
+                    throw new BroException("Idk what it means!");
+            }
+        }
+
+    public void checkEmptyInput(String input) throws BroException{
```

Comment:
Should there be a whitespace before the '{'? I noticed the same issue in several other places too. However, this should be solved easily once you implement CheckStyle 😉

Reply:
sure 
Thanks, I will make all these changes!!